caas/kubernetes/provider/provider.go

```diff
@@ -28,11 +28,24 @@ import (
 	"github.com/juju/juju/environs/context"
 )
 
+// ClusterMetadataChecker provides an API to query cluster metadata.
```

Comment:
All the interfaces moved from caas into caas/kubernetes/provider should probably be in caas/kubernetes instead.

Reply:
No problem. I will move those and also try to relocate some k8s constants (IIRC they are defined in the caas package) there.